Shop_Unity_base/Assets/Scripts/Shop/View/ShopMessageView.cs

```diff
@@ -2,24 +2,24 @@
 {
     using UnityEngine;
     using Model;
+    using Model.Items;
 
     public class ShopMessageView : MonoBehaviour
     {
         private ShopModel shopModel;
 
         // This method is used to initialize values, because we can't use a constructor.
-        public void Initialize(ShopModel shopModel)
+        public void Initialize(ShopModel shopModel, ITrader otherTrader)
         {
             this.shopModel = shopModel;
 
-            ShopModel.OnBuyEvent += OnShopUpdated;
-            ShopModel.OnSellEvent += OnShopUpdated;
+            RegisterEvents(otherTrader);
+            RegisterEvents(otherTrader);
```

Comment:
Why are there two calls to this? Did you mean to do this instead?
```suggestion
            RegisterEvents(this);
            RegisterEvents(otherTrader);
```

Reply:
Good catch